### ISSUE TITLE
Make Unique (Recursive) option is not present for an array of resources.

### ISSUE THREAD

**Issue by Isenther** [godotengine/godot#80100]
### Godot version

4.2 Dev 2

### System information

Windows 11, Intel(R) Core(TM) i5-10400F CPU, NVIDIA GeForce GTX 1060 6GB, 24 ram

### Issue description

![image (2)](https://github.com/godotengine/godot/assets/51789223/006d6f20-d797-4186-9bd5-b9fc10ee122b)

So i was duplicating my buttons and assigning new shortcuts to the new ones, but there is no "make unique (recursive)" option, so each time i ended up needing to go and turn both shortcut and InputEventKey unique.

### Steps to reproduce

No need.

### Minimal reproduction project

No need.

**Comment by dougVanny** [comment]
This is true not only when trying to make an array unique, but also when trying to make a resource that contains an array unique. After duplicating something I'll press the "Make Unique (Recursive)" believing that it will make the whole resource unique, but since it does not affect arrays, I end up modifying resources from the original copy that I did not intent to
